### ISSUE TITLE
Selecting local Blazor components in Razor LSP editor results in silent InvalidOperationException

### ISSUE THREAD

**Issue by NTaylorMullen** [dotnet/aspnetcore#21298]
- Disable Just My Code
- Turn on all `Common Language Runtime Exceptions`
![image](https://user-images.githubusercontent.com/2008729/80537636-3c00c700-8959-11ea-94d5-383ec502c202.png)
- Attach to another VS

In the "another VS"
- Open a Blazor app.
- Open Index.razor
- Type `<`
- Select the `App` completion item (don't have to commit it)
- See an InvalidOperationException get thrown:
![image](https://user-images.githubusercontent.com/2008729/80537489-04921a80-8959-11ea-99e3-c594100fdaaf.png)

Technically nothing bad happens as a result of selecting the completion item (vs doesn't explode or anything); however, it still throws which is super disruptive.

**Comment by ryanbrandenburg** [comment]
This is, unfortunately, expected. I've run into the same problem, here's what's happening.

As you can see from your callstack SumConverter from the VSLanguageServerClient is the last non-Newtonsoft parent of this. The way SumConverter decides which of its types is "correct" is that it just tries to Deserialize all of them, if one throws it tries the next one, if it succeeds then it decides that was the correct result. That means that for any given SumType Deserialization (of which CompletionItems are one) it is expected that exceptions are thrown.

So this doesn't represent a "bug" in that there is undesired behavior, but maybe we want to fix it anyway because this kind of control flow can make debugging a real pain.

**Comment by NTaylorMullen** [comment]
Not quite sure who owns the type but are we able to change `SumType` on their side? If not, would creating our own `SumType` be a good idea in the future, meaning would there be any benefits?

**Comment by ryanbrandenburg** [comment]
I'm currently looking at changing the implementation of SumConverter to handle `SumType` conversions in an exception-less way using JSONSchemas. I feel like I'm making good progress on that, though it's certainly more fiddly than the current implementation, but if it proves more troublesome than I thought or if it doesn't work I'll probably seek advise from our resident JSON expert.

**Comment by TanayParikh** [comment]
@ryanbrandenburg would this benefit from the `blocked` tag?

**Comment by ryanbrandenburg** [comment]
Yeah that's probably appropriate at this point. Basically my attempt to us JSONSchemas was pretty messy and fragile. The fix from their end would likely be when they move off of JSON serializing these things at all and into... @NTaylorMullen was it RPC?
